src/toil/mpcopy.py

```diff
@@ -0,0 +1,62 @@
+from urlparse import urlparse
+import boto
+import math
+import itertools
+
+
+class S3File:
+    def __init__(self, url):
+        self.s3 = boto.connect_s3()
+        self.url = urlparse(url)
+        assert self.url.scheme == 's3'
+        self.key = self.url.path[1:]
+        bucketName = self.url.netloc
+        self.bucket = self.s3.get_bucket(bucketName)
+
+    def copyTo(self, dest, totalSize, part_size=None, headers={}):
+        partSize = part_size or 2**20 * 50  # 50MiB
+
+        # initiate copy
+        upload = dest.bucket.initiate_multipart_upload(dest.key, headers=headers)
+        try:
+            start = 0
+            partIndex = itertools.count()
+            while start < totalSize:
+                end = min(start + partSize-1, totalSize-1)  # bytes are indexed form zero
```

Comment:
I always think and code with half-open intervals (end exclusive) and zero-based indices. When I interface with code that doesn't follow that convention I do the conversion as late as possible, e.g. ` copy_part_from_key(...,end=end-1)`.